### ISSUE TITLE
POSIX separation not complete

### ISSUE THREAD

**Issue by roblatham00** [LLNL/UnifyFS#663]
### System information

Type                   | Version/Name
---                    |     ---
Operating System       |  Linux (Ubuntu-20.04.02
UnifyFS Version        |   dev (v0.1.0-618-gc307f13c)

### Describe the problem you're observing

Unable to link against `libunifyfs_api` alone

### Describe how to reproduce the problem

consider this simple test case:

```c
#include <unifyfs/unifyfs_api.h>

#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>

#include <string.h>

#include <stdio.h>

int main(int argc, char **argv)
{
    unifyfs_cfg_option uopts;

    unifyfs_rc urc;
    unifyfs_handle uhandle;
    unifyfs_gfid ufid;

    urc = unifyfs_initialize(".", &uopts, 0, &uhandle);
    urc = unifyfs_create(uhandle, O_CREAT|O_RDWR, argv[1], &ufid);

    char *source_str = "Hello, Unify world!\n";
    char dest_str[128];
    unifyfs_io_request ureqs[2] =  {
        {
        .user_buf = source_str,
        .nbytes = strlen(source_str),
        .gfid = ufid,
        .op = UNIFYFS_IOREQ_OP_WRITE
        },
        {
        .user_buf = dest_str,
        .nbytes = strlen("world"),
        .offset = 13,
        .gfid = ufid,
        .op = UNIFYFS_IOREQ_OP_READ
        }
    };

    urc = unifyfs_dispatch_io(uhandle, 2, ureqs);
    if (strcmp(dest_str, "world") != 0) {
        fprintf(stderr, "unify i/o test fails");
    }
    unifyfs_finalize(uhandle);
}
```

Not doing anything tricky, right?

How should one compile this using the Unify API (not posix) ?

If I compile it like this: `gcc -I/home/robl/work/soft/unify/include rdwr.c -o rdwr -L/home/robl/work/soft/unify/lib -lunifyfs_api`

I get undefined symbols:

```
/usr/bin/ld: /home/robl/work/soft/unify/lib/libunifyfs_api.so.1: undefined reference to `posix_client_fini'
/usr/bin/ld: /home/robl/work/soft/unify/lib/libunifyfs_api.so.1: undefined reference to `client_rank'
/usr/bin/ld: /home/robl/work/soft/unify/lib/libunifyfs_api.so.1: undefined reference to `posix_client_init'
/usr/bin/ld: /home/robl/work/soft/unify/lib/libunifyfs_api.so.1: undefined reference to `global_rank_cnt'
/usr/bin/ld: /home/robl/work/soft/unify/lib/libunifyfs_api.so.1: undefined reference to `unifyfs_mount_id'
```

those symbols are defined in `posix_client.c` but if I add that file to the libunifyfs_api library I also bring in all the ``__real_fstatfs'` etc routines as well.

**Comment by MichaelBrim** [comment]
I can reproduce this using current dev and gcc10. I'm curious why this wasn't found by our unit test program that only links against libunifys_api - maybe because it's using libtool in -static mode?

**Comment by MichaelBrim** [comment]
Also, I'd just like to note that a read request will not see data from a write request in the same dispatch. The read needs to be in a second dispatch for the test code to work.

**Comment by MichaelBrim** [comment]
Closing, fixed in #665
